src/components/frontend-engine/event/context-provider.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, ReactElement, useRef } from "react";
+
+/**
+ * events are added / removed / dispatched via the eventManager
+ * the eventManager is just a div, the handling of events is purely through native event handlers
+ */
+interface IEventContext {
+	eventManager: Element;
+}
+
+interface IProps {
+	children: ReactElement;
+}
+
+export const EventContext = createContext<IEventContext>({
+	eventManager: null,
+});
+
+export const EventProvider = ({ children }: IProps) => {
+	const eventManager = useRef<Element>(document?.createElement("div"));
```

Comment:
instead of creating an element, there seems to be an event target api https://developer.mozilla.org/en-US/docs/Web/API/EventTarget

Reply:
ah oops. it's not available in older browsers like Safari 13 https://caniuse.com/mdn-api_eventtarget_eventtarget